pkg/multicluster/clientset.go

```diff
@@ -0,0 +1,81 @@
+package multicluster
+
+import (
+	"context"
+	"sync"
+
+	"github.com/rotisserie/eris"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+// ClientSet provides access to a client.Client for any registered cluster.
+// Only one ClientSet should be created within a multicluster system.
```

Comment:
nit: in Kubernetes, a `clientset` usually refers to a set of REST clients for an API group.

i would suggest `Client` here, wdyt?